### PR TITLE
Side 449 sidekick api needs to import ld

### PR DESCRIPTION
**Pull request by JonBernardOlive**
Adding these enums to the LDK so we can more seamlessly pass data between sidekick and sidekick-api

### CODE REVIEW

File: access.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"encoding/json"
+	"strings"
+)
+
+// Access is a grouping of entity type
+type Access int
+
+const (
+	// AccessUnknown is the access value used when the value is not known
+	AccessUnknown Access = iota
```
Comment (corylanou) [inline access.go:13]: Are these values ever passed outside this system or serialized?  If so, I would avoid `iota` and use actual constants. Just by changing the order of `iota` will change the value, and could result in corrupted data at a future point.

File: access.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	// AccessPublic is the value that allows any user of Sidekick to access the plugin
+	AccessPublic
+)
+
+func (a Access) String() string {
+	return [...]string{"unknown", "user", "organization", "public"}[a]
+}
+
+// MarshalJSON returns a json serialization of access values
+func (a Access) MarshalJSON() ([]byte, error) {
```
Comment (corylanou) [inline access.go:30]: Oh, there's my answer... so I would favor actual constant values over iota in this scenario.

**Comment by corylanou** [comment]
Being the new person, I'm unclear what our status is in regards to tests, so unless I'm told otherwise, I'll call out any time I think we should have some basic test coverage but don't.  I'm happy to write some tests for this as well. It would be quick and ensure that our serialization doesn't ever change in the future unexpectedly.  Thoughts?

File: access.go (hunk 1 above)
Comment (BillyMurrayOlive) [reply access.go:13]: This is my fault. JB is basing this on how I setup the "category" constant.
Given that I have already made this mistake for "category", is it better to correct it here and have them be inconsistent until we update category, or to keep everything consistent. I am leaning towards doing this one as you suggest and accepting the temporary inconsistency.

**Comment by BillyMurrayOlive** [comment]
@corylanou Our status is that tests are great, but we have not been doing them. 

This is my fault. I have not made testing a priority because we were (until recently) uncertain how permanent any of this work would be. Wanted to focus on getting the concept out to test ASAP. I know, bad code practices always have good intentions, it is still a mistake, just providing some context.

Please continue to call out tests when needed or add cards to the backlog if you see important places we need testing.

**Comment by BillyMurrayOlive** [comment]
@JonBernardOlive If we're ditching iota, you can remove most of the code. For example `operatingSystem.go` will just become:
```
package ldk

import (
	"strings"
)

// OperatingSystem is a grouping of entity type
type OperatingSystem string

const (
	// OperatingSystemUnknown is the OS value used when the OS is not known
	OperatingSystemUnknown OperatingSystem = "unknown"

	// OperatingSystemWindows is the OS value used for Windows
	OperatingSystemWindows OperatingSystem = "win32"

	// OperatingSystemMacOS is the OS value used for MacOS
	OperatingSystemMacOS OperatingSystem = "darwin"

	// OperatingSystemLinux is the OS value used for Linux
	OperatingSystemLinux OperatingSystem = "linux"

	// OperatingSystemAny is the OS value used a plugin is available on any OS
	OperatingSystemAny OperatingSystem = "any"
)
```

**Comment by JonBernardOlive** [comment]
*eyeroll*

File: operatingSystem.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	// OperatingSystemWindows is the OS value used for Windows
+	OperatingSystemWindows OperatingSystem = "win32"
+
+	// OperatingSystemMacOS is the OS value used for MacOS
+	OperatingSystemMacOS OperatingSystem = "darwin"
+
+	// OperatingSystemLinux is the OS value used for Linux
+	OperatingSystemLinux OperatingSystem = "linux"
+
+	// OperatingSystemAny is the OS value used a plugin is available on any OS
```
Comment (BillyMurrayOlive) [inline operatingSystem.go:19]: I think this is missing a "by".
`OperatingSystemAny is the OS value used by a plugin is available on any OS`

**Pull request merged**